lib/system/io.nim

```diff
@@ -798,10 +798,13 @@ when declared(stdout):
         when defined(windows):
           writeWindows(stdout, s)
         else:
-          discard c_fwrite(s.cstring, cast[csize_t](s.len), 1, stdout)
+          if c_fwrite(s.cstring, cast[csize_t](s.len), 1, stdout) != s.len:
```

Comment:
can you add a test similar to tests/exception/t9657.nim ?

eg:
```nim
echo "foo1"
close stdout
doAssertRaises(IOError):
  echo "foo"
``

Reply:
Yep... I was also thinking I probably need to put the whole thing in a try/finally block so that the lock releases happen even when there's an error.